app/src/main/java/mozilla/lockbox/presenter/RoutePresenter.kt

```diff
@@ -94,10 +94,19 @@ class RoutePresenter(
     }
 
     private fun showDialogFragment(dialogFragment: DialogFragment?) {
+        var src = navController.currentDestination ?: return
         if (dialogFragment != null) {
             val fragmentManager = activity.supportFragmentManager
             try {
                 dialogFragment.show(fragmentManager, dialogFragment.javaClass.name)
+                if (src.id == R.id.fragment_locked) {
```

Comment:
Rather than doing this logic in the `RoutePresenter`, I'd implement a second `RouteAction` to complement the `RouteAction.FingerprintDialog` that you already have. The `SettingPresenter` can be explicit about what dialog it's routing to rather than just the generic case. You can look at how I implemented the generic `DialogAction` class as an example.